test/testdata/cfg/dealias_with_return.rb.cfg.exp

```diff
@@ -55,7 +55,7 @@ subgraph "cluster_::Object#a" {
     "bb::Object#a_8" -> "bb::Object#a_6" [style="bold"];
     "bb::Object#a_9" [
         color = black;
-        label = "block[id=9, rubyBlockId=0](a: Integer(2))\l<statTemp>$14: Integer(3) = 3\l<returnMethodTemp>$2: Integer = a: Integer(2).+(<statTemp>$14: Integer(3))\l<finalReturn>: T.noreturn = return <returnMethodTemp>$2: Integer\l<unconditional>\l"
+        label = "block[id=9, rubyBlockId=0](a: Integer(2))\l<statTemp>$14: Integer(3) = 3\l<returnMethodTemp>$2: Integer = a: Integer(2).+(<statTemp>$14: Integer(3))\l<returnMethodTemp>$2: T.noreturn = return <returnMethodTemp>$2: Integer\l<finalReturn> = return <returnMethodTemp>$2\l<unconditional>\l"
```

Comment:
You'll notice two `return` instructions, one after another here. This happens because `builder_entry` unconditionally adds a final return, after synthesizing all instructions. Even on master, there will sometimes be cases where the CFG has a "double return" like this, if the user had written

```ruby
def foo
  return 0
end
```

directly. The second return is a stop-gap in case there was no explicit return from the method (e.g., an `if` that only had a `then` branch and an empty `else` branch).

Because `builder_entry` uses `synthesizeExpr` to call the make `return`, it is ignored by dead code checking, and thus ignored by inferrence.

Reply:
... actually: this is a bug 😅 it manifested here but also in the `suggest-sig` test. I've since fixed the bug, which makes this comment obsolete.